custom/icds_reports/middleware.py

```diff
@@ -48,12 +49,17 @@ def is_icds_dashboard_view(request):
 
 class ICDSAuditMiddleware(MiddlewareMixin):
     def process_view(self, request, view_func, view_args, view_kwargs):
-        if is_icds_dashboard_view(request):
+        if settings.SERVER_ENVIRONMENT in settings.ICDS_ENVS and is_icds_dashboard_view(request):
             ICDSAuditEntryRecord.create_entry(request)
             return None
 
     def process_response(self, request, response):
-        if is_login_page(request) and request.user.is_authenticated and is_icds_domain(request):
+        should_audit = (
+            settings.SERVER_ENVIRONMENT in settings.ICDS_ENVS and
```

Comment:
I think ICDS_ENVS is only the icds production envs? Right now we do a lot of ICDS QA on india, so I'm hesitant to make QA subtly different than the production environments.

though this is probably low risk

Reply:
> I think ICDS_ENVS is only the icds production envs? Right now we do a lot of ICDS QA on india, so I'm hesitant to make QA subtly different than the production environments.

Isn't this statement kind of in conflict with itself? If what you're saying is true then any functionality that is dependent on `ICDS_ENVS` would already be subdly different. I would think the better solution to the problem you're pointing out is to turn `ICDS_ENVS` on for india (though admittedly I don't know the implications of that)

Reply:
Hmm. That's a fair point. I briefly searched around for the usages of this and it doesn't look like we'd want to add india to ICDS_ENVS.

I added what I think is currently the best solution for this, though tbh I'm not super happy with it.

If the icds database is defined then we include the middleware, otherwise we skipp the middleware entirely.

Another option would be including this in LOCAL_MIDDLEWARE in localsettings.py on certain envs. However the way I implemented doesn't require any ansible localsettings deploys so it feels a bit more contained. What do you think?